base/linalg/dense.jl

```diff
@@ -351,6 +354,45 @@ function inv{T}(A::StridedMatrix{T})
     return convert(typeof(AA), Ai)
 end
 
+# Computes inverses for 1x1, 2x2, 3x3 matrices
+function small_inv{T}(A::StridedMatrix{T})
+    n = chksquare(A)
+    d = det(A)
+    if d == zero(T)
```

Comment:
I think it is fine to throw for exact zeros only. That is similar to what LAPACK does.